onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -2280,6 +2300,19 @@ bool TensorrtExecutionProvider::DetectTensorRTGraphCycles(SubGraphCollection_t&
 std::vector<std::unique_ptr<ComputeCapability>>
 TensorrtExecutionProvider::GetCapability(const GraphViewer& graph,
                                          const IKernelLookup& /*kernel_lookup*/) const {
+  // Construct subgraph capability from node list
+  std::vector<std::unique_ptr<ComputeCapability>> result;
+
+  // If the model consists of only a single "EPContext" contrib op, it means TRT EP can fetch the precompiled engine info from the node and
+  // load the engine directly without having to go through the processes of graph proto reconstruction, calling TRT parser and engine compilation.
+  // So, simply return the ComputeCapability here.
+  if (graph.NumberOfNodes() == 1 && GraphHasCtxNode(graph)) {
+    SubGraph_t supported_node_vector = {{0}, true};
```

Comment:
Wouldn't the condition to bypass the builder rather be that all nodes are EP context nodes ? In theory we could have an ONNX file with multiple EP Context nodes. It's probably not possible currently but it could even be multiple engines connected with control flow nodes that are executed on CPU ?

Reply:
yes, we are in the process of designing multiple context node support. (this is needed for other EP's as well)
For now, can we explicitly enforce that we can only handle a single EP context node here?
If there is more than one, we need to throw exception?